src/main/java/de/hhu/propra/sharingplatform/service/ItemService.java

```diff
@@ -64,6 +71,7 @@ public Item findItem(long itemId) {
 
     public void editItem(Item newItem, long oldItemId, long userId) {
         Item oldItemRental = findItem(oldItemId);
+        ItemValidator.validateItemIsFree(offerRepo, contractRepo, oldItemRental);
```

Comment:
Validation should also take place on trying to open the edit view and not only on storing changes

Reply:
fixed